pulp_ansible/app/downloaders.py

```diff
@@ -84,8 +84,8 @@ async def _run(self, extra_data=None):
         :meth:`~pulpcore.plugin.download.BaseDownloader._run`.
 
         """
-        if not self.token:
-            # No Token
+        if not self.ansible_auth_url:
```

Comment:
If there is no auth_url, but there is a token I believe we need to use that. I think that would be a call to _run_with_additional_headers because the token is the extra Authorization header that is needed.

If there is no token and no auth_url then call return await super()._run(extra_data=extra_data).

If there is both a token and auth_url call self._run_with_token_refresh_and_401_retry()